src/persist-client/src/lib.rs

```diff
@@ -453,6 +456,25 @@ impl PersistClient {
         Ok(reader)
     }
 
+    /// Creates and returns a [BatchFetcher] for the given shard id.
+    #[instrument(level = "debug", skip_all, fields(shard = %shard_id))]
+    pub fn create_batch_fetcher<K, V, T, D>(&self, shard_id: ShardId) -> BatchFetcher<K, V, T, D>
+    where
+        K: Debug + Codec,
+        V: Debug + Codec,
+        T: Timestamp + Lattice + Codec64,
+        D: Semigroup + Codec64 + Send + Sync,
+    {
+        let fetcher = BatchFetcher {
```

Comment:
how would we feel about something like calling `StateVersions::fetch_current_state` or `StateVersions::maybe_init_shard` here as a mechanism to detect misuse. that would read from the network, but it wouldn't serialize on anything (or write to state)

Reply:
to ensure the codecs match

Reply:
@guswynn I added this, as a check, so we don't need to rename to `unchecked_` and add the safety warning. Would you agree?